### PR TITLE
Language change issue 

### PR DESCRIPTION
**Pull request by addegbenga**
This PR fixes an issue where attempting to change the language from English to any other triggers a "Document not found (404)" error [#763](https://github.com/cairo-book/cairo-book/issues/763)

- - -
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/840)

### CODE REVIEW

**Comment by TAdev0** [comment]
thanks for the quick PR @addegbenga  . Were you able to test these changes?

**Comment by addegbenga** [comment]
> thanks for the quick PR @addegbenga . Were you able to test these changes?

Yeah sure I did a quick deployment to make sure it works.

**Comment by TAdev0** [comment]
so the missing`/html` was the key! thanks!

**Review by TAdev0 — APPROVED** [review]

File: .github/workflows/mdbook.yml [hunk 1]
```diff
@@ -44,23 +44,28 @@ jobs:
       - name: Install mdbook
         uses: ./.github/workflows/install-mdbook
 
-      - name: Build book (English)
-        run: mdbook build -d book
+      - name: Build all Books
```
Comment (enitrat) [inline .github/workflows/mdbook.yml:47]: Thank you, that looks good (moving book/lang/html to book/html/lang, now that we have book/html and book/markdown directories, this was required)

Can you just keep the file as it was before (One step is Build book, the other step is Build all translations) ? There should only be one line changed which is `mv book/$po_lang/html book/html/$po_lang`

Comment (addegbenga) [reply .github/workflows/mdbook.yml:47]: Alright will do that

**Review by TAdev0 — APPROVED** [review]

**Comment by TAdev0** [comment]
thanks @addegbenga !

**Comment by TAdev0** [comment]
@addegbenga  i just checked and it seems only some chapters or pages are indeed translated, do you have the same issue on your side?

**Comment by addegbenga** [comment]
> @addegbenga i just checked and it seems only some chapters or pages are indeed translated, do you have the same issue on your side?

Yes, I'm seeing the same issue on my end too. I'm not sure why it's happening, but I can do a quick check. It doesn't seem related to the build, so I can try to resolve it in a separate PR.

**Comment by TAdev0** [comment]
would be great, thanks ! it seems the same .md files are not translated for all langages

**Comment by addegbenga** [comment]
> would be great, thanks ! it seems the same .md files are not translated for all languages

I conducted a quick check last night and found issues with some of the po.xx files:
1. Missing translations.
2. Syntax errors in some po.xx files, causing certain markdown text to not be translated properly.

For example, I tested the es.po file and observed these issues in the attached image. As a result, some pages are translated while others are not.

**Tips for Fixing the Issues**

1. Review and Correct: Carefully review the po.xx files to fix all errors and add any missing translations.
2. Automate Formatting: Implement a script to format po.xx files automatically in the CI pipeline.
3. Use Translation Tools: Advise contributors to use tools like Poedit to review translations to catch errors early(I think this is already in readme).
4. Write a Guide: Create a proper guide on testing translations locally before making a pull request.

I think these tips can be worded better and  be opened for contribution to the community.


Screenshot 2024-06-23 at 10 15 32

**Comment by TAdev0** [comment]
thanks for investigating @addegbenga 

I noticed its always the same chapters that are not translated for all languages, do you know why? I mean, the creation of new files for new langages is automatic right?

**Comment by addegbenga** [comment]
> thanks for investigating @addegbenga
> 
> I noticed its always the same chapters that are not translated for all languages, do you know why? I mean, the creation of new files for new langages is automatic right?

There are two main reasons why this happen:

1. The sections in po.xx file for that chapter are not properly translated.
2. The translation for some sections in that chapter is missing.

Yes, file creation for new languages is automatic when you build.

I can't confirm that this issue happens for the same chapters in all languages. from my observation, when I switch to different languages such as Spanish (es) and French (fr), I notice a mix of translated and untranslated sections.
You can check Array chapter and switch between es and fr to see):

**Comment by addegbenga** [comment]
In the first image you see that the second paragraph was not translated. 

I used Poedit to fix this, resulting in the corrected version in the second screenshot on my local machine. 
Thats mostly the issue we are facing for all the chapters.

Screenshot 2024-06-23 at 11 53 36
Screenshot 2024-06-23 at 11 53 55

**Comment by TAdev0** [comment]
@addegbenga  how can we solve this so that just running the tool will work ? any idea why the translation tool itself doesnt work properly?

**Pull request merged**